llvm/lib/Transforms/IPO/AttributorAttributes.cpp

```diff
@@ -2902,6 +2902,22 @@ struct AANonConvergentImpl : public AANonConvergent {
   }
 };
 
+static bool FunctionCalledWithConvergenceToken(const Function *F) {
+  for (auto &Use : F->uses()) {
+    CallBase *CB = dyn_cast<CallBase>(Use.getUser());
+    if (!CB)
+      continue;
+
+    // We are not called, just used as an argument.
+    if (CB->getCalledFunction() != F)
+      continue;
+
+    if (CB->getConvergenceControlToken())
+      return true;
+  }
+  return false;
```

Comment:
This needs to be a conservatively correct true. If the address is captured you do not know the users. If the function is externally visible, you also do not know the users.